Enable pocl builds on macOS

jakirkham opened the pull request:
Replaces PR ( https://github.com/conda-forge/clblast-feedstock/pull/1 ).

Configures the custom matrix for `pocl` builds and enables them on macOS.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham committed:
Add OSX_VARIANT matrix configuration

Configure the `OSX_VARIANT` matrix to also contain the `pocl` variant in
addition to the `native` configuration on macOS. Both only are ever
defined on macOS.

jakirkham committed:
MNT: Re-rendered with conda-smithy 2.4.5

Add `OSX_VARIANT` to the Travis CI matrix. Only try two values for it
under macOS, the already present `native` configuration and now the
`pocl` configuration. On all other platforms, `OSX_VARIANT` is simply
ignored.
